### PR TITLE
FTR 1037 workspace progress chart

### PR DESCRIPTION
**Pull request by RazgrizX**
Closes (1037 board)
Added a chart to the user profile in guider and chart for every workspace.
Gets data from Muikku database and MongoDB
Uses back-end restpoints and controllers
Uses new addition to currentuser load action
Adds statistics property to the redux store (guider.currentUser.statistics)
Uses AMchart library
Contains commented parts and marks
Only a prototype at the moment
Works in mobile
Uses muikku-icons/1.2.4

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java [hunk 1]
```diff
@@ -2694,4 +2695,28 @@ public Response deleteWorkspaceFileContent(@PathParam("WORKSPACEID") Long worksp
         .build();
   }
   
+  @GET
+  @Path("/workspaces/{WORKSPACEENTITYID}/activityStatistics")
+  @RESTPermit(handling = Handling.INLINE)
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java]: Inline permissions but not checked. Should probably check that logged user can check their own statistics or if the logged user has some specific permission (workspace teacher? some high-up environment roles?).

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/dao/WorkspaceMaterialReplyDAO.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import javax.persistence.criteria.CriteriaQuery;
 import javax.persistence.criteria.Root;
 
+import fi.otavanopisto.muikku.model.users.UserEntity;
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/dao/WorkspaceMaterialReplyDAO.java]: Unused import?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java (hunk 1 above)
Comment (RazgrizX) [reply muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java]: Permission check added

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application/current-student/filters/graph-filter.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+class GraphFilter extends React.Component<GraphFilterProps> {
+  constructor(props:GraphFilterProps) {
+    super(props);
+  }
+  
+  dropdownFilter(){
+    let modificator = this.props.modificator || "";
+    if (modificator == "-dropdown-only" || modificator == ""){
+      return <Dropdown modifier={"graph-filter" + modificator} persistant={true} items={this.props.graphs.map((graph)=>{
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application/current-student/filters/graph-filter.tsx]: Very minor thing :) it's totally okay but you can actually just do `persistant` without `persistant={true}` and it's exactly the same; but it's okay, you don't have to change it, it's just the way it's commonly done.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java (hunk 1 above)
Comment (onzag) [reply muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java]: Where?....

Comment (onzag) [reply muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java]: Oh yes sorry, this didn't display all the lines... I think you should poke viljakainen to resolve this.

**Pull request merged**